### PR TITLE
[WIP, far from finished] Remove unnecessary includes

### PR DESCRIPTION
**Pull request by practicalswift**
Remove unnecessary includes.

_WIP: This PR is *far* from finished. Submitting this version to trigger a Travis CI build to get testing on multiple platforms. Let's discuss which of these includes that should still be kept around despite being "technically redundant" and which we really want to remove._

### CODE REVIEW

**Comment by sipa** [comment]
NACK.

You're removing includes of direct dependencies because they're already included indirectly. That results in fragile code that breaks when indirect dependencies change.

In general, a .cpp and .h file should `#include` every header that it directly uses function/classes/definitions from, even if that's already implied by other dependencies. One exception is that a .cpp file does not need to re-include things that its corresponding .h already included.

**Comment by practicalswift** [comment]
This PR is far from complete - it is a rough first version. Updated PR description.

I'll try to find the subset of these includes that are not only "technically redundant" but also unnecessary in the sense that they don't reflect use.

Let me know if you see any such instances right away from the list.

**Comment by practicalswift** [comment]
I look into setting up my own Travis CI account so that I don't pollute the main project with WIP PR:s to trigger Travis. Sorry about that!

File: src/base58.cpp [hunk 1]
```diff
@@ -4,16 +4,8 @@
 
 #include "base58.h"
 
-#include "hash.h"
-#include "uint256.h"
 
-#include <assert.h>
-#include <stdint.h>
 #include <string.h>
-#include <vector>
```
Comment (luke-jr) [inline src/base58.cpp]: `std::vector` is used in this file, so the include is necessary. I imagine this is true of many other removed includes...

Comment (practicalswift) [inline src/base58.cpp]: Yes, that's true! Please disregard this WIP PR (it shouldn't have been submitted in the first place - I should have found another way to trigger Travis CI testing), see the final PR "Remove includes in .cpp files for things the corresponding .h file already included" instead :-)

Sorry about taking up reviewer time with this WIP stuff.

**Pull request closed without merging**